// react-qr-code/lib/components/QRCodeSurfaceCell/index.native-and-web.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = function (Shape, Transform) {
  var propTypes = {
    d: _react2.default.PropTypes.string.isRequired,
    fill: _react2.default.PropTypes.string.isRequired,
    transformX: _react2.default.PropTypes.number.isRequired,
    transformY: _react2.default.PropTypes.number.isRequired
  };

  var defaultProps = {};

  var QRCodeSurfaceCell = function QRCodeSurfaceCell(_ref) {
    var d = _ref.d,
        fill = _ref.fill,
        transformX = _ref.transformX,
        transformY = _ref.transformY;

    var transform = new Transform().translate(transformX, transformY);
    return _react2.default.createElement(Shape, {
      fill: fill,
      transform: transform,
      d: d
    });
  };

  QRCodeSurfaceCell.propTypes = propTypes;
  QRCodeSurfaceCell.defaultProps = defaultProps;

  return QRCodeSurfaceCell;
};
